java/ql/src/experimental/Security/CWE/CWE-352/JsonpInjectionLib.qll

```diff
@@ -0,0 +1,114 @@
+import java
+import DataFlow
+import JsonStringLib
+import semmle.code.java.security.XSS
+import semmle.code.java.dataflow.DataFlow
+import semmle.code.java.dataflow.DataFlow3
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.frameworks.spring.SpringController
+
+/**
+ * A method that is called to handle an HTTP GET request.
+ */
+abstract class RequestGetMethod extends Method {
+  RequestGetMethod() {
+    not exists(MethodAccess ma |
+      ma.getMethod() instanceof ServletRequestGetBodyMethod and
+      any(this).polyCalls*(ma.getEnclosingCallable())
```

Comment:
`any(this)` doesn't make sense; what do you want to achieve here and why?

Reply:
Ensure that the get request method will not process the request body.

Reply:
This processing is necessary. Is my writing wrong?

Reply:
Could you give an example of a case that this should filter out?

Reply:
https://github.com/leiyi1152/shancai/blob/master/src/main/java/com/icloud/web/user/app/AppUserController.java#L50

[readToJSONObect](https://github.com/leiyi1152/shancai/blob/941ad9037b80551d46060f2986cb0bb268b80d1c/src/main/java/com/icloud/common/util/RequestUtil.java#L35) function will call `getInputStream()` to get the request body.

Reply:
OK, in that case just `any(this)` -> `this`